src/cmd/cli/command/mcp.go

```diff
@@ -17,13 +18,21 @@ import (
 var mcpCmd = &cobra.Command{
 	Use:   "mcp",
 	Short: "Manage MCP Server for defang",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		//set global nonInteractive to false
+		nonInteractive = false
+	},
 }
 
 var mcpServerCmd = &cobra.Command{
 	Use:   "serve",
 	Short: "Start defang MCP server",
 	Args:  cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		authPort, err := cmd.Flags().GetInt("auth-server")
+		if err != nil {
+			return err
+		}
```

Comment:
We don't bother anywhere else. 
```suggestion
		authPort, _err_ := cmd.Flags().GetInt("auth-server")
```